cmakelib/lexer/lexer_test.go

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright 2019 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package lexer
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/alecthomas/participle/lexer"
+	plex "github.com/alecthomas/participle/lexer"
+	"github.com/alecthomas/repr"
+)
+
+type Token = plex.Token
+
+func NewToken(kind rune, value string) Token {
+	return Token{Type: kind, Value: value}
+}
+
+func NewTokenAt(kind rune, value string, offset, line, col int) Token {
+	return Token{Type: kind,
+		Value: value,
+		Pos: lexer.Position{
+			Filename: "",
+			Offset:   offset,
+			Line:     line,
+			Column:   col,
+		},
+	}
+}
+
+func compareTokens(a, b []Token) bool {
```

Comment:
You might find https://godoc.org/github.com/google/go-cmp/cmp helpful here, and I believe we already depend on it.

Reply:
Thanks! This is much better.